lib/Conversion/TritonGPUToLLVM/TritonGPUToLLVM.cpp

```diff
@@ -883,6 +884,39 @@ struct LoadOpConversion
       const int vecNElems = totalWidth / valueElemNbits;
       assert(wordNElems * nWords * numVecs == numElems);
 
+#ifdef USE_ROCM
+      GCNBuilder gcnBuilder;
+
+      const std::string readConstraint = "v";
+      const std::string writeConstraint = "=v";
+
+      auto *dstsOpr = gcnBuilder.newListOperand();
+      for (size_t wordIdx = 0; wordIdx < nWords; ++wordIdx) {
+        auto *opr = gcnBuilder.newOperand(writeConstraint); // =v operations
+        dstsOpr->listAppend(opr);
+      }
+
+      auto *addrOpr = gcnBuilder.newAddrOperand(ptrElems[vecStart], "v");
+
+      for (size_t wordIdx = 0; wordIdx < nWords; ++wordIdx) {
+        auto &gload =
+            gcnBuilder.create<GCNMemInstr>("global_load")->type(width);
```

Comment:
In GCN, there is V4 version of load/store. Will this width reflect that?

Reply:
In the current version memory load instructions are not vectorized. I plan to do the optimizations a little later